src/lib.rs

```diff
@@ -232,11 +233,10 @@ where
 
     /// Iterate over ReadType objects.
     fn next(&mut self) -> Option<Self::Item> {
-        match self.read_pair_iter.next() {
-            Some(Ok(read)) => Some(Ok(self.processor.process_read(read))), // Processed Read
-            Some(Err(e)) => Some(Err(e.into())),                           // IO Error
-            None => None,                                                  // End of fastq
-        }
+        self.read_pair_iter.next().map(|rr| {
+            rr.map(|read| self.processor.process_read(read))
+                .map_err(FastqError::into)
+        })
```

Comment:
I feel like it was more cleaner the old way, perhaps because of the comments.

Reply:
agree to disagree?  This is [generally considered more idiomatic](https://rust-lang.github.io/rust-clippy/master/#manual_map)

Reply:
Fair enough!